include/ipc/ipc_service.h

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright (c) 2021 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_IPC_SERVICE_IPC_SERVICE_H_
+#define ZEPHYR_INCLUDE_IPC_SERVICE_IPC_SERVICE_H_
+
+#include <stdio.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief IPC Service API
+ * @defgroup ipc_service_api IPC service APIs
+ * @{
+ */
+
+/** @brief Event callback structure.
+ *  It is registered during endpoint registration.
+ *  This structure is part of the endpoint configuration.
+ */
+struct ipc_service_cb {
+	/** @brief Bind was successful.
+	 *  @param priv Private user data.
+	 */
+	void (*bound)(void *priv);
+
+	/** @brief New packet arrived.
+	 *  @param data Pointer to data buffer.
+	 *  @param len Length of @a data.
+	 *  @param priv Private user data.
+	 */
+	void (*received)(void *data, size_t len, void *priv);
+
+	/** @brief An error occurred.
+	 *  @param message Error message.
+	 *  @param priv Private user data.
+	 */
+	void (*error)(const char *message, void *priv);
+};
+
+/** @brief Endpoint instance.
+ *  Content is not important for user of the API.
+ *  It is implemented in a specific backend.
+ */
+struct ipc_ept;
+
+/** @brief Endpoint configuration structure. */
+struct ipc_ept_cfg {
+	const char *name;           /**< @brief Name of the endpoint. */
```

Comment:
I think you shouldn't use a `brief` for structure member. Also you can reconsider comments formatting. For example:
```suggestion
        /** Name of the endpoint. */
	const char *name;
	
	/** Id of the target processor.*/
	int target_core_id; 
	
```